lte/gateway/python/magma/pipelined/qos/tc_ops_pyroute2.py

```diff
@@ -0,0 +1,134 @@
+"""
+Copyright 2021 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+
+from .tc_ops import TcOpsBase
+import logging
+from pyroute2 import IPRoute, NetlinkError
+import pprint
+
+LOG = logging.getLogger('pipelined.qos.tc_pyroute2')
+
+QUEUE_PREFIX = '1:'
+PROTOCOL = 0x0800
+
+
+class TcOpsPyRoute2(TcOpsBase):
+    """
+    Create TC scheduler and corresponding filter
+    """
+    def __init__(self):
+        self._ipr = IPRoute()
+        self._iface_if_index = {}
+        LOG.info("initialized")
+
+    def create_htb(self, iface: str, qid: str, max_bw: int, rate: str,
+                   parent_qid: str = None) -> int:
+
+        try:
+            LOG.debug("rate %s mac bw %s", rate, max_bw)
+            if_index = self._get_if_index(iface)
+            htb_queue = QUEUE_PREFIX + qid
+            ret = self._ipr.tc("add-class", "htb", if_index,
+                               htb_queue, parent=parent_qid,
+                               rate=str(rate).lower(), ceil=max_bw, prio=1)
+            LOG.debug("Return: %s", ret)
```

Comment:
Return from tc add subclass

Reply:
yeah I just meant expand log msg so its clear

Reply:
logger has prefix set. So in log it looks like:
```
Mar  1 21:59:15 magma-dev pipelined[11667]: INFO:pipelined.qos.tc_pyroute2:initialized
```